spring-integration-smb/src/test/java/org/springframework/integration/smb/session/SmbSessionTests.java

```diff
@@ -58,7 +58,7 @@ public void testCreateSmbFileObjectWithBackSlash3() throws IOException {
 		SmbShare smbShare = new SmbShare("smb://myshare\\shared\\");
 		SmbSession smbSession = new SmbSession(smbShare);
 		SmbFile smbFile = smbSession.createSmbFileObject("..\\another");
-		assertEquals("smb://myshare/another/", smbFile.getCanonicalPath());
+		assertEquals("smb://myshare/shared/../another", smbFile.getCanonicalPath());
```

Comment:
This is something wrong, too.
If we request to create a file in the root of the `smb://myshare`, then we need to pursue that anyway.
What happened with relative path resolution in this new client?

Reply:
Same comment as above.

Reply:
Yes, please, address that concern for better reflection of reality.

Thanks

Reply:
Ok I've update the SmbSessionTests to use getPath() instead of getCanonicalPath(). It looks much better and appears to assert the data that this test was originally checking for.